test/widget/nav_bar/widget/nav_bar_container_test.dart

```diff
@@ -0,0 +1,359 @@
+          expect(stateNavBar.config, equals(_backBtnConfig));
+          // we need code above to make sure the config was not null
+
+          // now we will return an ignored config instead of backBtn
+          firstTimeCall = false;
+          resetNavBar();
+          await tester.pumpAndSettle(updateNabBarDebounceTimeout);
+          expect(stateNavBar.config, isNull);
+          // we want to verify that ignored `config` was ignored and new config is null
+
+          // wait until NavBar update finished
+          await tester.pumpAndSettle();
+        },
+      );
+      testWidgets(
+        'GIVEN configMixins list with 2 items WHEN _updateBar method called THEN forEach happened in reversed way and latest item in the list will apply',
+        // the latest widget in the widget tree with the NavBarConfigMixin will be applied
+        (final WidgetTester tester) async {
+          final widget = _StatelessConfigWidget(
+            () => _singleItemConfig,
+            child: _StatelessConfigWidget(() => _backBtnConfig),
+          );
+          await tester.pumpLindenApp(buildWidget(child: widget));
+
+          final stateNavBar = getStateNavBar();
+
+          expect(stateNavBar.config, isNull);
+
+          resetNavBar();
+          await tester.pumpAndSettle(updateNabBarDebounceTimeout);
+
+          expect(stateNavBar.config, _backBtnConfig);
+        },
+      );
+    },
+  );
+
+  group(
+    'static methods',
+    () {
+      Future<BuildContext> buildTreeWithoutNavBarContainer(
+        WidgetTester tester,
+      ) async {
+        await tester.pumpLindenApp(const Center());
+        return find.byType(Center).evaluate().first;
+      }
+
+      testWidgets(
+        'GIVEN widget tree without NavBarContainer WHEN calling reset and update NavBar THEN throw exception',
+        (final WidgetTester tester) async {
+          final context = await buildTreeWithoutNavBarContainer(tester);
+
+          expect(
+            () => resetNavBar(context: context),
+            throwsA(isA<NavBarContainerNotFoundException>()),
+          );
+          expect(
+            () => resetNavBar(context: context, goingBack: true),
+            throwsA(isA<NavBarContainerNotFoundException>()),
+          );
+          expect(
+            () => updateNavBar(context: context),
+            throwsA(isA<NavBarContainerNotFoundException>()),
+          );
+        },
+      );
+      testWidgets(
+        'GIVEN staticCallsEnabled == false WHEN calling reset and update NavBar THEN nothing happened',
+        (final WidgetTester tester) async {
+          final context = await buildTreeWithoutNavBarContainer(tester);
+
+          NavBarContainer.staticCallsEnabled = false;
+
+          resetNavBar(context: context);
+          resetNavBar(context: context, goingBack: true);
+          updateNavBar(context: context);
+
+          NavBarContainer.staticCallsEnabled = true;
+        },
+      );
```

Comment:
❓ This test seems to only test a test relevant function and contains 2 of three instances where we set `staticCallsEnabled` maybe we could remove this test and find another solution for the one remaining call?

Reply:
discussed, can be resolved